fpdf/fpdf.py

```diff
@@ -3427,6 +3415,7 @@ def _render_styled_text_line(
                         self._resource_catalog.add(
                             PDFResourceType.FONT, current_font.i, self.page
                         )
+                    current_font_is_added = True
```

Comment:
Shoudn't that be reset to False at tome point?

What happens when a text fragment with a different font is parsed?
Is it correctly added to the content stream?